src/services/azureBlobStorageService.ts

```diff
@@ -0,0 +1,141 @@
+  }
+
+  /**
+   * Upload a file to Azure Blob Storage
+   * @param path Path of file to upload
+   * @param containerName Name of container in Azure Blob storage for upload
+   * @param blobName Name of blob file created as a result of upload
+   */
+  public async uploadFile(path: string, containerName: string, blobName?: string) {
+    const name = blobName || path.replace(/^.*[\\\/]/, "");
+    uploadFileToBlockBlob(Aborter.none, path, this.getBlockBlobURL(containerName, name));
+  };
+  
+  /**
+   * Delete a blob from Azure Blob Storage
+   * @param containerName Name of container containing blob
+   * @param blobName Blob to delete
+   */
+  public async deleteFile(containerName: string, blobName: string): Promise<void> {
+    const blockBlobUrl = await this.getBlockBlobURL(containerName, blobName)
+    await blockBlobUrl.delete(Aborter.none);
+  }
+
+  /**
+   * Lists files in container
+   * @param ext - Extension of files to filter on when retrieving files
+   * from container
+   */
+  public async listFiles(containerName: string, ext?: string): Promise<string[]> {
+    const result: string[] = [];
+    let marker;
+    const containerURL = this.getContainerURL(containerName);
+    do {
+      const listBlobsResponse = await containerURL.listBlobFlatSegment(
+        Aborter.none,
+        marker,
+      );
+      marker = listBlobsResponse.nextMarker;
+      for (const blob of listBlobsResponse.segment.blobItems) {
+        if ((ext && blob.name.endsWith(ext)) || !ext) {
+          result.push(blob.name);
+        }
+      }
+    } while (marker);
+
+    return result;
+  }
+
+  /**
+   * Lists the containers within the Azure Blob Storage account
+   */
+  public async listContainers() {
+    const result: string[] = [];
+    let marker;
+    do {
+      const listContainersResponse = await this.getServiceURL().listContainersSegment(
+        Aborter.none,
+        marker,
+      );
+      marker = listContainersResponse.nextMarker;
+      for (const container of listContainersResponse.containerItems) {
+        result.push(container.name);
+      }
+    } while (marker);
+
+    return result;
+  }
+
+  /**
+   * Creates container specified in Azure Cloud Storage options
+   * @param containerName - Name of container to create
+   */
+  public async createContainer(containerName: string): Promise<void> {
+    const containerURL = this.getContainerURL(containerName);
+    await containerURL.create(Aborter.none);
+  }
+
+  /**
+   * Delete a container from Azure Blob Storage Account
+   * @param containerName Name of container to delete
+   */
+  public async deleteContainer(containerName: string): Promise<void> {
+    const containerUrl = await this.getContainerURL(containerName)
+    await containerUrl.delete(Aborter.none);
+  }
+
+  /**
+   * Get ServiceURL object for Azure Blob Storage Account
+   */
+  private getServiceURL(): ServiceURL {
+    const credential = this.credentials
```

Comment:
Any reason not to use this.credentials directly vs split out into separate variable?

Reply:
Nope :)